File: ResourceManager/Models/Configurations/ArticleTagConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class ArticleTagConfiguration : IEntityTypeConfiguration<ArticleTag>
    {
        public void Configure(EntityTypeBuilder<ArticleTag> builder)
        {
            builder.ToTable("ArticleTags");
            builder.HasKey(x => new { x.ArticleID, x.TagID });

            builder.HasOne(x => x.Article)
                .WithMany(x => x.ArticleTags)
                .HasForeignKey(x => x.ArticleID);

            builder.HasOne(x => x.Tag)
                .WithMany(x => x.ArticleTags)
                .HasForeignKey(x => x.TagID);

        }
    }
}

File: ResourceManager/Models/Configurations/ArticleTagConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore

Namespace ResourceManager.Models.Configurations
    Public Class ArticleTagConfiguration
        Implements IEntityTypeConfiguration(Of ArticleTag)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of ArticleTag)) Implements IEntityTypeConfiguration(Of ArticleTag).Configure
            builder.ToTable("ArticleTags")
            builder.HasKey(Function(x) New With {x.ArticleID, x.TagID
            })

            builder.HasOne(Function(x) x.Article).WithMany(Function(x) x.ArticleTags).HasForeignKey(Function(x) x.ArticleID)

            builder.HasOne(Function(x) x.Tag).WithMany(Function(x) x.ArticleTags).HasForeignKey(Function(x) x.TagID)

        End Sub
    End Class
End Namespace
